pkg/statistics/handle/autoanalyze/priorityqueue/calculatoranalysis/calculator_analysis_test.go

```diff
@@ -0,0 +1,356 @@
+	"github.com/pingcap/tidb/pkg/statistics/handle/types"
+	"github.com/stretchr/testify/require"
+)
+
+// Constants for test thresholds and file names
+const (
+	inputDataFile = "data.csv"
+	goldenFile    = "calculated_priorities.golden.csv"
+)
+
+// Constants for generating test data
+const (
+	maxChangeRate       = 0.001 // 0.1% per second maximum change rate
+	maxChangePercentage = 3.0   // Maximum change capped at 300% of table size
+	minMeaningfulChange = 1000  // Minimum meaningful change amount
+)
+
+// Constants for change rate calculation
+const (
+	baseChangeRate      = 0.001  // Base change rate: 0.1% per second
+	changeRateDecayLog  = 3      // Controls how quickly the change rate decays for larger tables
+	smallTableThreshold = 100000 // Tables smaller than this use the base change rate
+)
+
+var update = flag.Bool("update", false, "update .golden files")
+
+// Please read README.md for more details.
+// To update golden file, run the test with -update flag.
+func TestPriorityCalculatorWithCSVData(t *testing.T) {
+	generateTestData(t)
+	jobs, err := loadTestData(inputDataFile)
+	require.NoError(t, err, "Failed to load test data")
+
+	calculator := priorityqueue.NewPriorityCalculator()
+
+	newPriorities := calculateNewPriorities(jobs, calculator)
+	sortPrioritiesByWeight(newPriorities)
+
+	// Convert newPriorities to CSV string
+	var buf bytes.Buffer
+	writer := csv.NewWriter(&buf)
+	writer.Write([]string{"ID", "CalculatedPriority", "TableSize", "Changes", "TimeSinceLastAnalyze", "ChangeRatio"})
+	for _, p := range newPriorities {
+		writer.Write([]string{
+			fmt.Sprintf("%d", p.job.ID),
+			fmt.Sprintf("%.4f", p.priority),
+			fmt.Sprintf("%.0f", p.job.TableSize),
+			fmt.Sprintf("%.0f", p.job.Changes),
+			fmt.Sprintf("%.0f", p.job.TimeSinceLastAnalyze),
+			fmt.Sprintf("%.4f", p.changeRatio),
+		})
+	}
+	writer.Flush()
+	got := buf.Bytes()
+
+	// Compare with golden file
+	golden := goldenFile
+	if *update {
+		t.Log("updating golden file")
+		err = os.WriteFile(golden, got, 0644)
+		require.NoError(t, err, "Failed to update golden file")
+	}
+
+	want, err := os.ReadFile(golden)
+	require.NoError(t, err, "Failed to read golden file")
+
+	require.Equal(t, want, got)
+}
+
+// calculateNewPriorities calculates priorities for all jobs
+func calculateNewPriorities(jobs []TestJob, calculator *priorityqueue.PriorityCalculator) []jobWithPriority {
+	newPriorities := make([]jobWithPriority, 0, len(jobs))
+	for _, job := range jobs {
+		priority := calculator.CalculateWeight(&job)
+		changeRatio := job.Changes / job.TableSize // Calculate change ratio
+		newPriorities = append(newPriorities, jobWithPriority{job, priority, changeRatio})
+	}
+	return newPriorities
+}
+
+// sortPrioritiesByWeight sorts jobs by priority in descending order
+func sortPrioritiesByWeight(priorities []jobWithPriority) {
+	sort.Slice(priorities, func(i, j int) bool {
+		return priorities[i].priority > priorities[j].priority
+	})
```

Comment:
```suggestion
	slices.SortStableFunc(priorities, func(i, j jobWithPriority) int {
		return cmp.Compare(j.priority, i.priority)
	})
```
Should we use stable sort here so that the result remains stable when multiple jobs have the same priority?

Reply:
Make sense. Updated.